### DISCUSSION TITLE
Using dpdata package to generate training set from CP2K2023 version

### DISCUSSION THREAD

**Discussion by StealFox97** [deepmodeling/dpdata#498] [Q&A] [answered]
Dear developer:
Dpdata-matser only supports the *.log file of cp2k6.1. But I still want to use the calculated results from CP2K2023.
 **I modified the source code which under the path of “../dpdata/cp2k/output.py”, in the dpdata package. I made the code in output.py be able to get the key word in the output file of cp2k2023. I uploaded the “output.py” file that i modified.
For "fmt = ‘cp2k/output’", it can work now!
However, for “fmt = ‘cp2k/aimd_output’”, it seems that the code can only extract the first frame of *.log file.

`import dpdata

data = dpdata.LabeledSystem('run', fmt='cp2k/aimd_output')
print(len(data))
1
`

[test.zip](https://github.com/deepmodeling/dpdata/files/12078941/test.zip)

I am poor at Python. Which source code should I modifie next?
(Output.py was uploaded in txt format below）
[output.txt](https://github.com/deepmodeling/dpdata/files/12078934/output.txt)

**Comment by StealFox97** [comment]
/

**Reply by robinzyb** [reply]
Please check out the devel branch of https://github.com/robinzyb/cp2kdata/tree/devel

**Comment by anmol-compchem** [comment]
@robinzyb can you please provide any script that I can run to get the desired output(i want to convert cp2k 2022.01 data to npy for deepmd kit?

**Reply by StealFox97** [reply]
Thank to the developer. The cp2kdata pakage is very esay to use. But I have some leisure time recently ,and i have modified the "output.py" script suuccesfuly. Now, it can work with output  file which product by cp2k2023. I think it can also work with cp2k2022.01 because the two cp2k version have same surcture of output file.
@anmol-compchem I can share it with you if you want to try. Remamber change the output file name to *.log
[output.py.txt.txt](https://github.com/deepmodeling/dpdata/files/12145036/output.py.txt.txt)

**Reply by robinzyb** [reply, marked as answer]
Hello, because I just relocated to a new place recently, I didn't reply in time. Recently, we also update support for cp2k 2023. I will rountinely check the issue now. https://github.com/link89/cp2kdata/tree/fix-2023-support

**Reply by th-nic** [reply]
Does it support cp2k 8.2? I cannot convert my aimd_output to DeePMD system.

**Reply by robinzyb** [reply]
it is supposed to support v8.2 if you have any problem in using cp2kdata, please open an issue on cp2kdata repository

**Comment by anmol-compchem** [comment]
Thankyou for your reply when i am running the script 
i am getting this error
Traceback (most recent call last):
  File "/home/anmol/work/deep_md/00.data/readdata.py", line 5, in <module>
    data = dpdata.LabeledSystem('test', fmt = 'cp2k/aimd_output') 
  File "/home/anmol/.local/lib/python3.10/site-packages/dpdata/system.py", line 281, in __init__
    self.from_fmt(
  File "/home/anmol/.local/lib/python3.10/site-packages/dpdata/system.py", line 318, in from_fmt
    return self.from_fmt_obj(load_format(fmt), file_name, **kwargs)
  File "/home/anmol/.local/lib/python3.10/site-packages/dpdata/system.py", line 1162, in from_fmt_obj
    data = fmtobj.from_labeled_system(file_name, **kwargs)
  File "/home/anmol/.local/lib/python3.10/site-packages/dpdata/plugins/cp2k.py", line 13, in from_labeled_system
    return tuple(Cp2kSystems(log_file, xyz_file, restart))
  File "/home/anmol/.local/lib/python3.10/site-packages/dpdata/cp2k/output.py", line 58, in __next__
    log_info_dict = self.handle_single_log_frame(next(self.log_block_generator))
  File "/home/anmol/.local/lib/python3.10/site-packages/dpdata/cp2k/output.py", line 311, in handle_single_log_frame
    float(line_list[3]) * AU_TO_EV_EVERY_ANG,
ValueError: could not convert string to float: '[a.u.]'
attaching all files
[new.zip](https://github.com/deepmodeling/dpdata/files/12146363/new.zip)

**Comment by PolyuWeldingSpock** [comment]
> Thankyou for your reply when i am running the script i am getting this error Traceback (most recent call last): File "/home/anmol/work/deep_md/00.data/readdata.py", line 5, in data = dpdata.LabeledSystem('test', fmt = 'cp2k/aimd_output') File "/home/anmol/.local/lib/python3.10/site-packages/dpdata/system.py", line 281, in **init** self.from_fmt( File "/home/anmol/.local/lib/python3.10/site-packages/dpdata/system.py", line 318, in from_fmt return self.from_fmt_obj(load_format(fmt), file_name, **kwargs) File "/home/anmol/.local/lib/python3.10/site-packages/dpdata/system.py", line 1162, in from_fmt_obj data = fmtobj.from_labeled_system(file_name, **kwargs) File "/home/anmol/.local/lib/python3.10/site-packages/dpdata/plugins/cp2k.py", line 13, in from_labeled_system return tuple(Cp2kSystems(log_file, xyz_file, restart)) File "/home/anmol/.local/lib/python3.10/site-packages/dpdata/cp2k/output.py", line 58, in **next** log_info_dict = self.handle_single_log_frame(next(self.log_block_generator)) File "/home/anmol/.local/lib/python3.10/site-packages/dpdata/cp2k/output.py", line 311, in handle_single_log_frame float(line_list[3]) * AU_TO_EV_EVERY_ANG, ValueError: could not convert string to float: '[a.u.]' attaching all files [new.zip](https://github.com/deepmodeling/dpdata/files/12146363/new.zip)

Hi,

Do you solve the problem? I met same issue.

**Reply by StealFox97** [reply]
I am sorry. I am poor at Python. I have bee find long time, and i can not solve this problem. I also got this error when i use your file. But it is work normally with cp2k2023 version in my computer.

**Reply by anmol-compchem** [reply]
there is some issue with 2022.1 output but its working well for cp2k2023 output

**Comment by PolyuWeldingSpock** [comment]
> there is some issue with 2022.1 output but its working well for cp2k2023 output

Did you replace the output.py in dpdata/cp2k/output.py with the newfile mentioned above?
